### ISSUE TITLE
Unexpected cursor artifacts when hidden window is focused

### ISSUE THREAD

**Issue by errael** [neovim/neovim#30553]
### Problem

This is about having a visible cursor in a non-focused area; it is a visual artifact. This happens with both a text window and the command line.

This is a follow on issue after #30503 is fixed. IMHO, it seems low priority.

This might be considered an artificial situation. It's possible to enter hidden windows through scripts (or command line). Even though, in general, a script shouldn't leave the focus in a hidden window, there's some behavior I'm seeing that might be considered a bug.

Note: this issue is independent of whether or not the hidden window is focusable.

### Steps to reproduce

1. Edit the test file below, move the cursor to the middle of the file.
2. `:sou` creating the hidden focusable window
3. Execute the mapping `Z2`. Moves the focus to the hidden window
   Does `vim.api.nvim_set_current_win(popup_wid)` 

Observe: Cursor is visible in the main/**non-focused** window.

4. Scroll the window with the mouse so only last line of buffer is visible in the window.

Observe the cursor doesn't move and ends up away from the buffer's text lines.

5. Execute some ":" commands

Observe the cursor move around and get stuck in the command line.

```lua
local popup_wid
local function Pop1Any()
    local bnr = vim.api.nvim_create_buf(false, true)
    assert(bnr, "Failed to create buffer")
    vim.api.nvim_buf_set_lines(bnr, 0, -1, true, {'simple', 'win'})

    vim.api.nvim_set_option_value("bufhidden", "wipe", {buf = bnr})
    vim.api.nvim_set_option_value("modifiable", true, {buf = bnr})
	popup_wid = vim.api.nvim_open_win(bnr, false, {
		relative = "editor",
		style = "minimal",
		width = 10,
		height = 2,
		focusable = false,
		hide = true,
		col = 10,
		row = 10,
	})
end

Pop1Any()

vim.keymap.set('n', 'Z1', function() vim.print(vim.inspect(vim.api.nvim_list_wins())) end)
vim.keymap.set('n', 'Z2', function() vim.api.nvim_set_current_win(popup_wid) end)
vim.keymap.set('n', 'Z3', ":call nvim_set_current_win(" .. popup_wid .. ")<CR>")
vim.keymap.set('n', 'Z4', function() vim.api.nvim_win_set_config(popup_wid, {hide=false}) end)
vim.keymap.set('n', 'Z5', function() vim.api.nvim_win_set_config(popup_wid, {hide=true}) end)
vim.o.tm = 5000
```

### Expected behavior

Not sure. Maybe when hidden is focused, probably no visible cursor in regular; and for command line, only visible while entering a command.

### Nvim version (nvim -v)

NVIM v0.11.0-dev - https://github.com/neovim/neovim/commit/7b71fdbc1e9fcb71e642e67e0ac9a2711dd67df0

### Vim (not Nvim) behaves the same?

NA

### Operating system/version

ubuntu

### Terminal name/version

gnome

### $TERM environment variable

xterm-256color

### Installation

make install
